projects/rero/ng-core/src/lib/record/search/aggregation/slider/slider.component.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * RERO angular core
+ * Copyright (C) 2020 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { RecordSearchService } from '../../record-search.service';
+
+@Component({
+  selector: 'ng-core-aggregation-slider',
+  templateUrl: './slider.component.html'
+})
+export class AggregationSliderComponent implements OnInit {
+  // Aggregation key.
+  @Input()
+  key: string;
+
+  // Buckets list
+  @Input()
+  buckets: Array<any>;
+
+  // Min value.
+  @Input()
+  min = 1;
+
+  // Max value.
+  @Input()
+  max = 1000;
+
+  // Step
+  @Input()
+  step = 1;
+
+  // Range to search
+  range: Array<number> = null;
+
+  // True if route have aggregation query param.
+  hasQueryParam = false;
+
+  /**
+   * Constructor.
+   *
+   * @param _recordSearchService Record search service.
+   * @param _route Current route.
+   */
+  constructor(private _recordSearchService: RecordSearchService, private _route: ActivatedRoute) { }
+
+  /**
+   * Component initialization.
+   *
+   * Subscribe to route changes for getting aggregation query parameter.
+   */
+  ngOnInit() {
+    this._route.queryParams.subscribe(params => {
```

Comment:
Seems not consistent within the aggregations components. Can you use the record-search service?